code/controllers/subsystem/blackmarket.dm

```diff
@@ -113,3 +115,19 @@ SUBSYSTEM_DEF(blackmarket)
 		return FALSE
 	queued_purchases += P
 	return TRUE
+
+/// Used to repopulate the market when the auction rotation happens.
+/datum/controller/subsystem/blackmarket/proc/repopulate_market(market)
+	markets[market].available_items.Cut()
+	markets[market].categories.Cut()
+
+	for(var/item in subtypesof(/datum/blackmarket_item))
+		var/datum/blackmarket_item/I = new item()
+
+		if(!I.item)
```

Comment:
if I has no item var that sounds like a bug, it should return a report of it if thats the case.

Reply:
same here